apps/src/code-studio/viewAsRedux.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Reducer for tracking whether we're currently viewing the page as a student
+ * or as a teacher
+ */
+
+import { makeEnum } from '@cdo/apps/utils';
+import { queryParams, updateQueryParam } from '@cdo/apps/code-studio/utils';
+
+export const ViewType = makeEnum('Student', 'Teacher');
+
+// Action types
+export const SET_VIEW_TYPE = 'viewAs/SET_VIEW_TYPE';
+
+/**
+ * Stage lock reducer
+ */
+export default function reducer(state = ViewType.Student, action) {
+  if (action.type === SET_VIEW_TYPE) {
+    const viewType = action.viewType;
+    if (!ViewType[viewType]) {
+      throw new Error('unknown ViewType: ' + viewType);
```

Comment:
What, no test for the error case?

Reply:
My defense is that I was just moving existing tests/behavior, and it wasn't previously being tested. That's probably not a very good defense though, and now seems like a good time to rectify this